### ISSUE TITLE
Right click on script icon should allow to detach script

### ISSUE THREAD

**Issue by CubeOfCheese** [godotengine/godot-proposals#9534]
### Describe the project you are working on

Teaching my students how to use Godot

### Describe the problem or limitation you are having in your project

Sometimes the students accidentally attach a script to a node they weren't supposed to. I point it out to them by pointing to the script icon. "Look, there's a script attached to this node and it shouldn't be. Let's detach it."
They then try to right click on the script icon expecting to get a menu with the option to delete it.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Right clicking on the script icon should produce an options menu that allows the script to be detached.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![20240416_095841](https://github.com/godotengine/godot-proposals/assets/16740469/417aa8dc-b310-4aba-8135-7516de770292)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't think so

### Is there a reason why this should be core and not an add-on in the asset library?

It would make the engine a little more intuitive to interact with for all beginners

**Comment by KoBeWi** [comment]
There are already at least 3 ways to detach a script. Also this could lead to accidentally removing the script (it's less likely with right-click, but still).

**Comment by AThousandShips** [comment]
I agree that this would make it easy to accidentally detach it, especially since the assumption would be a right click would open a menu, and detaching a script is quite destructive, losing all data and exports

**Comment by AThousandShips** [comment]
Oh that makes more sense, the title of the proposal is very misleading

**Comment by timothyqiu** [comment]
Icons on the right side are technically buttons, so allowing right clicking feels unusual.

It's fine if we conceptually treat them as icons instead. But how should other icons react to right clicking? It's still bad if only the script icon acts differently.

**Comment by CubeOfCheese** [comment]
> But how should other icons react to right clicking? It's still bad if only the script icon acts differently.

I could imagine the scene icon having a right click menu that includes 'reveal in file system' for example
